caddyconfig/httpcaddyfile/directives.go

```diff
@@ -120,6 +120,17 @@ func RegisterHandlerDirective(dir string, setupFunc UnmarshalHandlerFunc) {
 	})
 }
 
+// RegisterGlobalOption registers a unique global option opt with
+// an associated unmarshaling (setup) function. When the global
+// option opt is encountered in a Caddyfile, setupFunc will be
+// called to unmarshal its tokens.
+func RegisterGlobalOption(opt string, setupFunc UnmarshalGlobalFunc) {
```

Comment:
Terminology question - these are called "global options" in the code, but the parallel non-global versions of these are called "directives"... Should this be `RegisterGlobalDirective`? Or does that have different semantics?

Reply:
That's more a Q for @mholt but I'm following the nomenclature as seen here https://caddyserver.com/docs/caddyfile/options

Reply:
Global options are for configuring things that are configured outside of an HTTP request / handler chain. If your directive just adds HTTP route(s), then you don't need/want global option.